python/graphstorm/model/node_gnn.py

```diff
@@ -169,15 +171,15 @@ def node_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
     device = model.device
     data = loader.data
     g = data.g
-    preds = []
+    preds = {ntype: [] for ntype in g.ntypes}
 
     if return_label:
         assert data.labels is not None, \
             "Return label is required, but the label field is not provided whem" \
             "initlaizing the inference dataset."
 
-    embs = []
-    labels = []
+    embs = {ntype: [] for ntype in g.ntypes}
+    labels = {ntype: [] for ntype in g.ntypes}
```

Comment:
is it a problem that the dict has all node types?

Reply:
instead of `target_ntypes`

Reply:
I updated this in the latest commit 582bb64e0703f6d5490529443bac56913777115f by 

- first create an empty `embs` as dictionary
- adding per node type `emb` to `embs` only if it is returned by the data loader

Since data loader only return node types in `target_ntypes`, it should be fine right now in this latest commit.